bot.py

```diff
@@ -338,6 +338,147 @@ async def display_current_list_of_summoners(ctx):
         await ctx.send(embed=create_embed(embed_data))
 
 
+# pylint: disable=too-many-locals, too-many-branches, too-many-statements
+@bot.command(name="remove", help="Remove player(s) from the list")
+async def remove_summoner(ctx, *, message):
+    """Remove summoner(s) from list
+    and send  the list to the bot"""
+
+    try:
+        # typing indicator
+        async with ctx.typing():
+            await asyncio.sleep(1)
+
+        # converting the message into list of summoners
+        summoner_to_remove_input = message.split(",")
+
+        # Exception case: attempt to remove more than 10 players
+        if len(summoner_to_remove_input) > MAX_NUM_PLAYERS_TEAM:
+            raise Exception (
+                "Limit Exceeded",
+                "You have exceeded a limit of 10 summoners! \
+                \nPlease remove {0} less summoners!".format(
+                    MAX_NUM_PLAYERS_TEAM - len(summoner_to_remove_input)
+                )
+            )
+
+        # Exception case: data/data.json file does not exist
+        if not os.path.exists(json_path):
+            raise Exception (
+                "Limit Exceeded",
+                "There is no summoner(s) added in the game.\nPlease add summoner(s) first!"
+            )
+
+        # for importing data from json file
+        file_data = ""
+        # initializing server id to a variable
+        server_id = str(ctx.guild.id)
+
+        if os.path.getsize(json_path) > 0:
+            with open(json_path, "r") as file:
+                file_data = json.load(file)
+        else:
+            raise Exception (
+                "Limit Exceeded",
+                "There is no summoner(s) added in the game.\nPlease add summoner(s) first!"
+            )
+
+        players_to_remove = []
+        summoners_to_remove = []
+        if server_id in file_data:
+            for player_name in summoner_to_remove_input:
+                for player in file_data[server_id]:
+                    if normalize_name(player_name) == normalize_name(player["user_name"]):
+                        players_to_remove.append(player)
+                        summoners_to_remove.append(player_name)
```

Comment:
why do you need two different variables; `players_to_remove`,`summoners_to_remove`  that basically stores the same thing?

Reply:
`players_to_remove` holds players such as
```
{
            "user_name": "soultree0",
            "formatted_user_name": "soultree0",
            "tier_division": "GOLD",
            "tier_rank_number": "IV"
        }
```
`summoners_to_remove` hold actual summer_name such as `soultree0`, `bongbong`, ...
So they are different.

But I guess for `summoners_to_remove`, I could just replace with `player["user_name"]`
I will make change to above.

Reply:
No actually, I can't.
`summoners_to_remove` is a summoner_name but at the same time, it's a user input. So it could come as "Bong Bong"
So unless, I want to normalize all of them in the beginning, and return message to customer with normalized "Bong Bong" which is "bongbong, I don't see other way.
<img width="567" alt="Screen Shot 2021-07-13 at 12 21 01 AM" src="https://user-images.githubusercontent.com/9527124/125390063-3762db80-e370-11eb-911e-cde107b75e34.png">
This is a behaviour I want to keep, with "Unregistered Summoner(s)" and returning whatever user input.

Reply:
okay - `player_name` should be renamed to 'summoner_input` this code is not very readable atm. 

players_to_remove and summoners_to_remove is just an empty list to start off.. 
what's wrong with putting `player_name` in one object together?
eg; player['player_name'] = player_name and append it to the list?

Reply:
@smoon91 
`player['player_name'] = player_name` won't work in a case where
- user's input player name --> "Bong Bong"
- player["user_name"] from file_data[server_id] --> "bongbong"

When attempting to remove with `summoner_to_remove_input.remove(player["username"])`
This will not work because:
- player["username"] --> `bongbong`
- `summoner_to_remove_input` --> ["Bong Bong", "coco and cookie", "Home Sweet", ... ]

Another way of making this work is to use another for-loop with normalizing `summoner_to_remove_input` but that will not satisfy https://github.com/dhmoon91/discord/pull/32#discussion_r668415665

Or maybe my brain is frozen right now and can't think straight.
Please feel free to correct me if something's wrong.

Reply:
not `summoner_to_remove_input.remove(player["username"])` summoner_to_remove_input.remove(player["player_name"]) ...

Reply:
issue handled by using pydash.find() function